Updated TOKEN_ISSUER to 'accounts.google.com'

arjun3396 opened the pull request:
Hi, I was getting this issue from today morning parse-server/Adapters/Auth/google.js was expecting the TOKEN_ISSUER to be prefixed with https:// but on debugging the original value was not having the prefix, removing https:// from TOKEN_ISSUER solved this bug. This issue is introduced in 4.3.0 as in 4.2.0 it is working fine currently I have downgraded the version to 4.2.0 for it to work properly and suggesting the changes please merge this PR.

dplewis commented:
The Travis Build is failing. Can you update the test cases?

dplewis commented:
@SebC99 Can you look at this? I see you worked on https://github.com/parse-community/parse-server/pull/6734

SebC99 commented:
As per Google official documentation, all examples are using `https` prefix, but it seems they have added this note: `The value of iss in the ID token is equal to accounts.google.com or https://accounts.google.com.`
So removing `https://` from the TOKEN_ISSUER will fail too if Google sends it in various format. 

In our cases, we always receive the `https://` version.
I suggest to test both versions, with and without https.

arjun3396 commented:
As per the discussion, I have added check for both https and without https and updated the test-cases please review or suggest if any other changes needed

arjun3396 commented:
> The Travis Build is failing. Can you update the test cases?

@dplewis I have updated the test case but this one below is failing, any suggestions on how to fix this?
 ParseGraphQLServer Auto API Schema Relay Spec Object Identification Id inputs should work either with global id or object id
  Message:
    Expected 'U2Vjb25kYXJ5T2JqZWN0OmVJWHBxZXZCUnY=' to be less than 'U2Vjb25kYXJ5T2JqZWN0Om9WN2Uxd09jdlY='.
  Stack:
    Error: Expected 'U2Vjb25kYXJ5T2JqZWN0OmVJWHBxZXZCUnY=' to be less than 'U2Vjb25kYXJ5T2JqZWN0Om9WN2Uxd09jdlY='.
        at <Jasmine>
        at UserContext.it_only_db (/home/travis/build/parse-community/parse-server/spec/ParseGraphQLServer.spec.js:2794:19)
        at process._tickCallback (internal/process/next_tick.js:68:7)

dplewis commented:
@arjun3396 Thats a flaky test that you can ignore.

RodrigoSMarques commented:
Hello guys. When will a new version of Parse Server be released with this fix?
I upgraded to version 4.3 but had to go back to 4.2 because login with Google didn't work anymore.

Thank you.
